packages/jazz-react-core/src/hooks.ts

```diff
@@ -26,16 +34,69 @@ export function useJazzContext<Acc extends Account>() {
   return value;
 }
 
+export function useJazzContextManager<Acc extends Account>() {
+  const value = useContext(JazzContextManagerContext) as JazzContextManager<
+    Acc,
+    {}
+  >;
+
+  if (!value) {
+    throw new Error(
+      "You need to set up a JazzProvider on top of your app to use this hook.",
+    );
+  }
+
+  return value;
+}
+
 export function useAuthSecretStorage() {
-  const value = useContext(JazzAuthContext);
+  const value = useContext(JazzContextManagerContext);
 
   if (!value) {
     throw new Error(
       "You need to set up a JazzProvider on top of your app to use this useAuthSecretStorage.",
     );
   }
 
-  return value;
+  return value.getAuthSecretStorage();
+}
+
+export function useIsAuthenticated() {
+  const authSecretStorage = useAuthSecretStorage();
+
+  return useSyncExternalStore(
+    useCallback(
+      (callback) => {
+        return authSecretStorage.onUpdate(callback);
+      },
+      [authSecretStorage],
+    ),
+    () => authSecretStorage.isAuthenticated,
+    () => authSecretStorage.isAuthenticated,
+  );
+}
+
+function useCoValueObservable<V extends CoValue, D>() {
+  const [initialValue] = React.useState(() =>
+    createCoValueObservable<V, D>({
+      syncResolution: true,
+    }),
+  );
+  const ref = useRef(initialValue);
+
+  return {
+    getCurrentValue() {
+      return ref.current.getCurrentValue();
+    },
+    getCurrentObservable() {
+      return ref.current;
+    },
+    reset() {
+      ref.current = createCoValueObservable<V, D>({
+        syncResolution: true,
+      });
```

Comment:
As a workaround for the issues related to the unsubscribe I've added a reset.

Going to fix the unsubscribe ASAP, but I think it's still better to fully isolate the subscribe actions with different observables.